src/compartment_sets.cpp

```diff
@@ -0,0 +1,670 @@
+
+
+#include "../extlib/filesystem.hpp"
+
+#include "utils.h"  // readFile
+
+#include <unordered_set>
+
+#include <bbp/sonata/compartment_sets.h>
+namespace bbp {
+namespace sonata {
+
+namespace fs = ghc::filesystem;
+
+namespace detail {
+
+using json = nlohmann::json;
+
+
+class CompartmentLocation
+{
+    private:
+      std::uint64_t node_id_;
+      std::uint64_t section_index_;
+      double offset_;
+
+      void setNodeId(int64_t node_id) {
+          if (node_id < 0) {
+              throw SonataError(fmt::format("Node id must be non-negative, got {}", node_id));
+          }
+          node_id_ = static_cast<uint64_t>(node_id);
+      }
+        void setSectionIndex(int64_t section_index) {
+            if (section_index < 0) {
+                throw SonataError(
+                    fmt::format("Section index must be non-negative, got {}", section_index));
+            }
+            section_index_ = static_cast<uint64_t>(section_index);
+        }
+        void setOffset(double offset) {
+            if (offset < 0.0 || offset > 1.0) {
+                throw SonataError(
+                    fmt::format("Offset must be between 0 and 1 inclusive, got {}", offset));
+            }
+            offset_ = offset;
+        }
+
+  public:
+    static constexpr double offsetTolerance = 1e-4;
+    // static constexpr double offsetToleranceInv = 1.0 / offsetTolerance;
```

Comment:
remove commented code?

Reply:
removed and removed the check with tolerance. Now we compare doubles... brrr

Reply:
tnx